### PR TITLE
Custom domain regex causing problems with servers (bugfix)

### PR DESCRIPTION
**Pull request by shults**
## Description

Other fixes in custom domain pattern that @radkrawczyk managed to find.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Refactoring or add test (improvements in base code or adds test coverage to functionality)

## Checklist

- [ ] I ensured that the documentation is up to date
- [ ] I explained why this PR updates go.mod in detail with reasoning why it's required
- [ ] I would like a code coverage CI quality gate exception and have explained why

___

### **PR Type**
Bug fix, Tests, Enhancement

___

### **Description**
Validate server variable names with regex
Validate and wrap invalid variable patterns
Add clearer error types and messages
Extend tests for invalid names/patterns

___

### Diagram Walkthrough

```mermaid
flowchart LR
  parser["serverUrlParser.extractValueBetweenBraces"] -- "validate name (nameRe)" --> nameCheck["ErrInvalidVariableName"]
  parser -- "compile pattern (regexp.Compile)" --> patCheck["ErrInvalidPattern via errpack.Wrap"]
  tests["servers_test"] -- "assert errors for edge cases" --> parser
  errpack["pkg/errpack"] -- "Wrap base errors with context" --> parser
```

<details> <summary><h3> File Walkthrough</h3></summary>

<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Enhancement</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>servers.go</strong><dd><code>Enforce variable name/pattern validation in server parser</code></dd></summary>
<hr>

internal/oasutil/servers.go

<ul><li>Add name validation with <code>nameRe</code> regex<br> <li> Compile and validate custom patterns using <code>regexp.Compile</code><br> <li> Introduce <code>ErrInvalidVariableName</code> and <code>ErrInvalidPattern</code><br> <li> Integrate <code>errpack.Wrap</code> for contextual error wrapping</ul>

</details>

  </td>
  <td><a href="https://github.com/TykTechnologies/tyk/pull/7310/files#diff-98dd06199bf9992e099563df9150f18cb38094f4dae3299f33c5330722ddac3d">+18/-0</a>&nbsp; &nbsp; </td>

</tr>

<tr>
  <td>
    <details>
      <summary><strong>errpack.go</strong><dd><code>Introduce simple error wrapping utility</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

pkg/errpack/errpack.go

<ul><li>Add lightweight error wrapping type<br> <li> Provide <code>Wrap</code> to prepend context to errors</ul>

</details>

  </td>
  <td><a href="https://github.com/TykTechnologies/tyk/pull/7310/files#diff-601b35aea34b6423c049e307a35817731433cc7ceec323f6f1118e4b2f260c44">+20/-0</a>&nbsp; &nbsp; </td>

</tr>
</table></td></tr><tr><td><strong>Tests</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>servers_test.go</strong><dd><code>Extend tests for invalid names and patterns</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

internal/oasutil/servers_test.go

<ul><li>Add tests for invalid variable name and pattern<br> <li> Ensure errors are non-nil and contain expected messages</ul>

</details>

  </td>
  <td><a href="https://github.com/TykTechnologies/tyk/pull/7310/files#diff-4a274e0f05feb520d7ff68fb48a7a28020ea7525820ac8105f4c898854a19af8">+3/-0</a>&nbsp; &nbsp; &nbsp; </td>

</tr>
</table></td></tr></tr></tbody></table>

</details>

___

### CODE REVIEW

File: internal/oasutil/servers.go [hunk 1]
```diff
@@ -160,3 +172,9 @@ func (p *serverUrlParser) nextParamName() string {
 	p.counter++
 	return fmt.Sprintf("%s%d", DefaultServerUrlPrefix, p.counter)
 }
+
+func isValidName(name string) bool {
+	return nameRe.MatchString(name)
+}
+
+var nameRe = regexp.MustCompile("^[a-zA-Z][a-zA-Z0-9_]*$")
```
Comment (MaciekMis) [inline internal/oasutil/servers.go]: Consider moving this var to the top of the file

Comment (shults) [reply internal/oasutil/servers.go]: ok

**Comment by pfatyga-codilime** [comment]
/dependency Could you please analyse this PR?
